diff --exclude='*.o' --exclude='*.so' -U4 /home/gserdyuk/gnucap/gnucap-work.github/gnucap-a/gnucap/models-bsim/BSIM3v330/b3ask.c ./b3ask.c
--- /home/gserdyuk/gnucap/gnucap-work.github/gnucap-a/gnucap/models-bsim/BSIM3v330/b3ask.c	2011-01-19 22:52:04.000000000 +0200
+++ ./b3ask.c	2011-01-29 01:00:32.000000000 +0200
@@ -110,9 +110,9 @@
         case BSIM3_VDS:
             value->rValue = *(ckt->CKTstate0 + here->BSIM3vds);
             return(OK);
         case BSIM3_CD:
-            value->rValue = here->BSIM3cd; 
+            value->rValue = here->BSIM3cd * here->BSIM3mode; 
             return(OK);
         case BSIM3_CBS:
             value->rValue = here->BSIM3cbs; 
             return(OK);
diff --exclude='*.o' --exclude='*.so' -U4 /home/gserdyuk/gnucap/gnucap-work.github/gnucap-a/gnucap/models-bsim/BSIM3v330/b3mask.c ./b3mask.c
--- /home/gserdyuk/gnucap/gnucap-work.github/gnucap-a/gnucap/models-bsim/BSIM3v330/b3mask.c	2011-01-19 22:52:04.000000000 +0200
+++ ./b3mask.c	2011-01-29 01:01:13.000000000 +0200
@@ -1239,8 +1239,14 @@
             return(OK);
         case BSIM3_MOD_KF:
             value->rValue = model->BSIM3kf;
             return(OK);
+            
+        case BSIM3_MOD_NMOS:            /*added to return device type  NMOS=1, PMOS=-1*/
+        case BSIM3_MOD_PMOS:
+            value->iValue = model->BSIM3type;
+            return(OK);
+            
         default:
             return(E_BADPARM);
     }
     /* NOTREACHED */
